src/database/hooks/useSaveAnonChatHook.ts

```diff
@@ -53,65 +85,38 @@ const useSaveAnonChatHook = () => {
 
     initAnonymousChat.message.cid = initAnonymousChat.message.cid.replace('messaging:', '');
 
-    const chat = ChatSchema.fromInitAnonymousChatAPI(initAnonymousChat, status);
-    await chat.save(localDb);
+    try {
+      await helperSaveChannel(initAnonymousChat, type);
+    } catch (e) {
+      console.log(e);
+    }
 
     try {
+      const chat = ChatSchema.fromInitAnonymousChatAPI(initAnonymousChat, status);
+      await chat.save(localDb);
       const members = originalMembers || object?.members;
-      members?.forEach((member) => {
-        const saveUserAndChannelListMember = async () => {
-          try {
-            const userMember = UserSchema.fromInitAnonymousChatAPI(
-              member,
-              initAnonymousChat?.message?.cid
-            );
-
-            await userMember.saveOrUpdateIfExists(localDb);
-          } catch (e) {
-            console.log('error saveChatFromOtherProfile userMember', e);
-          }
-          try {
-            const memberSchema = ChannelListMemberSchema.fromInitAnonymousChatAPI(
-              initAnonymousChat?.message?.cid,
-              uuid(),
-              member
-            );
-            memberSchema.saveIfNotExist(localDb);
-          } catch (e) {
-            console.log('error saveChatFromOtherProfile memberSchema', e);
-          }
-        };
-
-        Promise.all([saveUserAndChannelListMember()]);
+      const promises: Promise<void>[] = members?.map((member) => {
+        return saveUserAndChannelListMember(initAnonymousChat.message.cid, member);
       });
+
+      await Promise.all(promises);
     } catch (e) {
-      console.log('error saveChatFromOtherProfile');
       console.log(e);
+    } finally {
+      refresh('channelList');
+      refresh('chat');
+      if (withNavigate) {
+        helperGoToAnonymousChat(object, GROUP_INFO);
+      }
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this block of code, there are two separate try-catch blocks followed by a finally block. This structure can lead to confusion and potential issues with error handling. If an error occurs in the first try-catch block, the second one will still execute. If the intention is to stop execution on any error, consider combining these into a single try-catch-finally block.
```diff
-    try {
-      await helperSaveChannel(initAnonymousChat, type);
-    } catch (e) {
-      console.log(e);
-    }
-
-    try {
-      const chat = ChatSchema.fromInitAnonymousChatAPI(initAnonymousChat, status);
-      await chat.save(localDb);
-      const members = originalMembers || object?.members;
-      const promises: Promise<void>[] = members?.map((member) => {
-        return saveUserAndChannelListMember(initAnonymousChat.message.cid, member);
-      });
-
-      await Promise.all(promises);
-    } catch (e) {
-      console.log(e);
-    } finally {
-      refresh('channelList');
-      refresh('chat');
-      if (withNavigate) {
-        helperGoToAnonymousChat(object, GROUP_INFO);
-      }
-    }
+    try {
+      await helperSaveChannel(initAnonymousChat, type);
+
+      const chat = ChatSchema.fromInitAnonymousChatAPI(initAnonymousChat, status);
+      await chat.save(localDb);
+      const members = originalMembers || object?.members;
+      const promises: Promise<void>[] = members?.map((member) => {
+        return saveUserAndChannelListMember(initAnonymousChat.message.cid, member);
+      });
+
+      await Promise.all(promises);
+    } catch (e) {
+      console.error(e);
+      throw e;
+    } finally {
+      refresh('channelList');
+      refresh('chat');
+      if (withNavigate) {
+        helperGoToAnonymousChat(object, GROUP_INFO);
+      }
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->